DEV_ENVIRONMENT.md

```diff
@@ -0,0 +1,22 @@
+# Developer guide: getting your environment set up
+
+1. Make sure you have `node` installed with a version at _least_ 4.2.3.
+2. Run `npm install -g angular-cli` to install the Angular CLI.
+3. Fork the `angular/material2` repo. 
+4. Clone your fork. 
+   Reccomendation: name your git remotes `upstream` for `angular/material` 
+   and `<your-username>` for your fork. Also see the [team git shortcuts](https://github.com/angular/material2/wiki/Team-git----bash-shortcuts).
+5. From the root of the project, run `npm install`
+
+
+To build the project, run `ng build`. 
+To watch for changes and automatically rebuild, run `ng build --watch`
+
+To bring up a local server, run `ng serve`. This will automatically watch for changes and rebuild.
+After the changes rebuild, the browser currently needs to be manually refreshed.
```

Comment:
I figure we tell people to use the cli directly because that is meant to be the standard for angular apps, as opposed to whatever custom `npm` scripts individual projects might add.